### PR TITLE
Handled Baker & Account not persisted in the Db

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Fixes the issue 
https://github.com/Concordium/concordium-scan/issues/26

by catching all the errors if the Baker Or Account has not been persisted to database in the previous blocks

## Changes

Caches all the errors if the Baker Or Account has not been persisted to database in the previous blocks

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: backend/Application/Api/GraphQL/Import/BakerImportHandler.cs [hunk 1]
```diff
@@ -351,13 +359,19 @@ private async Task WorkAroundConcordiumNodeBug225(BlockInfo blockInfo, ImportSta
             if (txEvent is ConcordiumSdk.NodeApi.Types.BakerRemoved bakerRemoved)
             {
                 var pendingChange = await pendingChangeStrategy.SetPendingChangeOnBaker(bakerRemoved);
-                UpdateNextPendingBakerChangeTimeIfLower(pendingChange.EffectiveTime, importState);
+                if (pendingChange is not null)
+                {
+                    UpdateNextPendingBakerChangeTimeIfLower(pendingChange.EffectiveTime, importState);
+                }
```
Comment (abizjak) [inline backend/Application/Api/GraphQL/Import/BakerImportHandler.cs:365]: Do we not always expect the pending change to not be null?

Would it make sense to log the error then?

File: backend/Application/Api/GraphQL/Import/BakerImportHandler.cs [hunk 2]
```diff
@@ -351,13 +359,19 @@ private async Task WorkAroundConcordiumNodeBug225(BlockInfo blockInfo, ImportSta
...
             }
 
             if (txEvent is ConcordiumSdk.NodeApi.Types.BakerStakeDecreased stakeDecreased)
             {
                 var pendingChange = await pendingChangeStrategy.SetPendingChangeOnBaker(stakeDecreased);
-                UpdateNextPendingBakerChangeTimeIfLower(pendingChange.EffectiveTime, importState);
+                if (pendingChange is not null)
+                {
+                    UpdateNextPendingBakerChangeTimeIfLower(pendingChange.EffectiveTime, importState);
+                }
```
Comment (abizjak) [inline backend/Application/Api/GraphQL/Import/BakerImportHandler.cs:374]: Same here.

File: backend/Application/Api/GraphQL/Import/IPendingBakerChangeStrategy.cs [hunk 3]
```diff
@@ -92,13 +92,24 @@ public async Task<PendingBakerChange> SetPendingChangeOnBaker(BakerRemoved baker
...
 
     public async Task<PendingBakerChange> SetPendingChangeOnBaker(BakerStakeDecreased stakeDecreased)
     {
         var updatedBaker = await _writer.UpdateBaker(stakeDecreased, src => src.BakerId,
             (src, dst) => SetPendingChange(dst, src));
+
+        if (updatedBaker is null)
+        {
+            return null;
+        }
```
Comment (abizjak) [inline backend/Application/Api/GraphQL/Import/IPendingBakerChangeStrategy.cs:111]: Do we want to log something here? Do we not always expect a non-null baker?

File: backend/ConcordiumSdk/NodeApi/Types/JsonConverters/RewardStatusConverter.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public class RewardStatusConverter : JsonConverter<RewardStatusBase>
         return protocolVersion switch
         {
             null or 1 or 2 or 3 => JsonSerializer.Deserialize<RewardStatusV0>(ref reader, options)!,
-            4 => JsonSerializer.Deserialize<RewardStatusV1>(ref reader, options)!,
+            4 or 5 => JsonSerializer.Deserialize<RewardStatusV1>(ref reader, options)!,
```
Comment (abizjak) [inline backend/ConcordiumSdk/NodeApi/Types/JsonConverters/RewardStatusConverter.cs:15]: Isn't this part of the other PR already?

File: backend/CHANGELOG.md [hunk 5]
```diff
@@ -1,2 +1,3 @@
 ## Unreleased
-- Added support for CIS Token balances listing under accounts sidebar
\ No newline at end of file
+- Added support for CIS Token balances listing under accounts sidebar
+- Handeled exceptions when Baker & Account not available in Database while trying to update them.
```
Comment (abizjak) [inline backend/CHANGELOG.md:3]:
```suggestion
- Handled exceptions when Baker & Account not available in Database while trying to update them.
```

**Review by abizjak — APPROVED** [review]
As far as I see it the changes are mainly making the responses more robust, but they likely do not fix the underlying issue with database consistency.

However as we discussed in the meeting and on slack that will be done in a followup PR. So let's get this merged and tested as a temporary solution.

File: backend/Application/Api/GraphQL/Import/BakerImportHandler.cs (hunk 1 above)
Comment (parv0888) [reply backend/Application/Api/GraphQL/Import/BakerImportHandler.cs:365]: done

File: backend/Application/Api/GraphQL/Import/BakerImportHandler.cs (hunk 2 above)
Comment (parv0888) [reply backend/Application/Api/GraphQL/Import/BakerImportHandler.cs:374]: done

File: backend/Application/Api/GraphQL/Import/IPendingBakerChangeStrategy.cs (hunk 3 above)
Comment (parv0888) [reply backend/Application/Api/GraphQL/Import/IPendingBakerChangeStrategy.cs:111]: this is being handled by the caller of the method.

File: backend/ConcordiumSdk/NodeApi/Types/JsonConverters/RewardStatusConverter.cs (hunk 4 above)
Comment (parv0888) [reply backend/ConcordiumSdk/NodeApi/Types/JsonConverters/RewardStatusConverter.cs:15]: At all the other places. Protocol 5 is already included.

**Comment by SRConcordium** [comment]
@parv0888 , pagination is introduced now wherever there's more entries to show (transactions, tokens etc.) but observed couple of UI issues - 1) Specific gage numberings  are not there, so it's hard to know how many pages are there or which page the user is viewing currently and 2) Previous & Next buttons looks a single button (there should be space between these two buttons).
https://stagenet.ccdscan.io/?dcount=1&dentity=account&daddress=4ZXndsiRAtHFpKZeKpHS5xDyGrMHy9oYfCKQmf9VmRATx4GTzY

**Comment by SRConcordium** [comment]
Pagination_Tokens

**Pull request merged**